streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicManager.java

```diff
@@ -102,11 +97,21 @@ public void makeReady(final Map<String, InternalTopicConfig> topics) {
             final Set<NewTopic> newTopics = new HashSet<>();
 
             for (final InternalTopicConfig internalTopicConfig : topicsToBeCreated) {
-                final Properties topicProperties = internalTopicConfig.toProperties(windowChangeLogAdditionalRetention);
-                final Map<String, String> topicConfig = new HashMap<>(defaultTopicConfigs);
-                for (final String key : topicProperties.stringPropertyNames()) {
-                    topicConfig.put(key, topicProperties.getProperty(key));
+                final Map<String, String> topicConfig = internalTopicConfig.getProperties(defaultTopicConfigs, windowChangeLogAdditionalRetention);
+
+                if (topicConfig.containsKey(TopicConfig.SEGMENT_INDEX_BYTES_CONFIG) &&
```

Comment:
Can't we put this check into `StreamsConfig` ?

Reply:
I had tried that. The problem is that the default topic configs value are unknown at the beginning. But maybe I can rework something out of it.

Bare with me for another minor change.

Reply:
@mjsax Okay now I remembered why I cannot do that, because `ProducerConfig` constructor is not public in its package as `AdminClientConfig` constructor. That's why I cannot do sth. like

```
new ProducerConfig(getProducerConfigs("dummy")).getInt(ProducerConfig.BATCH_SIZE_CONFIG);
```

to read it out.

Reply:
Sounds like an inconsistency... Would it be a problem making those constructors (ie, also for consumer) public?